temporalio/lib/temporalio/client/async_activity_handle.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'temporalio/api'
+require 'temporalio/client/activity_id_reference'
+require 'temporalio/client/interceptor'
+require 'temporalio/error'
+
+module Temporalio
+  class Client
+    # Handle representing an external activity for completion and heartbeat. This is usually created via
+    # {Client.async_activity_handle}.
+    class AsyncActivityHandle
+      # @return [String, nil] Task token if created with a task token. Mutually exclusive with {id_reference}.
+      attr_reader :task_token
+
+      # @return [ActivityIDReference, nil] Activity ID reference if created with one. Mutually exclusive with
+      # {task_token}.
+      attr_reader :id_reference
+
+      # Create an async activity handle. {Client.async_activity_handle} is preferred over instantiating this directly.
+      def initialize(task_token_or_id_reference)
```

Comment:
```suggestion
      # @!visibility private
      def initialize(client:, task_token:, id_reference:)
```

Like other handles, this should never be instantiated directly (and therefore we can change up the constructor to be more like what we need). Then again this isn't really implemented yet, so it doesn't matter until then.